src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -477,6 +477,11 @@ private static ArgumentInfo GetArgumentInfo(Type type, object defaults)
                         throw Contracts.Except("Duplicate default argument '{0}' vs '{1}'", def.LongName, field.Name);
 
                     string name = ArgCase(field.Name);
+
+                    string attrName = null;
+                    if (attr.Name != null && attr.Name.ToLowerInvariant() != name.ToLowerInvariant())
```

Comment:
Do you care about empty strings too? If so, I would use  String.IsEmptyOrNull. 
Also you can do a case insensitive compare with String.Equals(a,b,StringComparison.OrginalIgnoreCase)

Reply:
(!string.IsNullorEmpty(attr.Name) && 
attr.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase)


---
In reply to: [248905796](https://github.com/dotnet/machinelearning/pull/2124#discussion_r248905796) [](ancestors = 248905796)

Reply:
Done

Reply:
See the impleemntation of `ArgumentAttribute.Name`; it already guards against this. If you tried to set something to be empty, or even all whitespace, the property value would be null.